crates/task-impls/src/helpers.rs

```diff
@@ -44,13 +44,13 @@ pub async fn broadcast_event<E: Clone + std::fmt::Debug>(event: E, sender: &Send
 /// Panics if the VID calculation fails, this should not happen.
 #[allow(clippy::panic)]
 pub async fn calculate_vid_disperse<TYPES: NodeType>(
-    txns: Vec<u8>,
+    txns: Arc<Vec<u8>>,
```

Comment:
```suggestion
    txns: &[u8],
```
Should work with just `&[u8]` no? I don't think it should ever have been a `Vec<u8>`. That was forcing a clone for each call to the function. 😬 It's a good thing we're cleaning this up.

Reply:
Yes, this is tricky with lifetimes because `txns` is passed into `spawn_blocking`.

Maybe there is another more elegant approach but keeping this as an `Arc<[u8]>` and just handing out a reference seems to work.